src/libstd/path.rs

```diff
@@ -525,6 +525,26 @@ impl<'a> Hash for PrefixComponent<'a> {
 ///
 /// See the module documentation for an in-depth explanation of components and
 /// their role in the API.
+///
+/// This `struct` is created from iterating over the [`path::Components`]
```

Comment:
The only way (that I know) to create a `Component` `struct` is by iterating over the `Components` `struct`.